src/reference/informative/informC04_component_reference.rst

```diff
@@ -8,4 +8,189 @@
 
   .. container:: infospec
 
-    **TODO**
+    .. container:: heading3
+
+      Understanding component references
+
+    As with references to units and variables, the term "component reference" refers to places where you need to specify a component by name.
+    This is done through the :code:`component_ref` attribute found in :code:`encapsulation` and :code:`import component` elements, as well as the :code:`component_1` and :code:`component_2` attributes of :code:`connection` elements.
+
+    The complicated part occurs during imports, and involves understanding the *scope* of a component's reference.
+    This is really just the appropriate name by which to call it depending on who is doing the calling.
+
+    Consider the following example.  
+    Here, two families live next door to one another, and within each family unit use the terms "husband" and "wife" to refer to one another.
+    In the wider neighbourhood, however, they must be referred to by their names: "BarbaraGood", "TomGood", "JerryLeadbetter", and "MargotLeadbetter".
+
+    .. code::
+
+      model: SurbitonNeighbourhood
+        ├─ component: TomGood <╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴┐
+        ├─ component: BarbaraGood <╴╴╴╴╴╴╴╴╴┐ ╷
+        ├─ component: MargotLeadbetter <╴╴╷ ╷ ╷
+        └─ component: JerryLeadbetter <╴┐ ╷ ╷ ╷
+                                        ╷ ╷ ╷ ╷
+                                 imported components
+                                        ╵ ╵ ╵ ╵
+      # In LeadbetterFamily.cellml:     ╵ ╵ ╵ ╵ 
+      model: LeadbetterFamilyModel      ╵ ╵ ╵ ╵
+        ├─ component: husband ╴╴╴╴╴╴╴╴╴╴┘ ╵ ╵ ╵
+        └─ component: wife ╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴┘ ╵ ╵
+                                            ╵ ╵
+      # In GoodFamily.cellml:               ╵ ╵
+      model: GoodFamilyModel                ╵ ╵
+        ├─ component: wife ╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴┘ ╵
+        └─ component: husband ╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴┘
```

Comment:
Replace with:
```
model: SurbitonNeighbourhood
  ├─ component: TomGood <───────────────┐
  ├─ component: BarbaraGood <─────────┐ │
  ├─ component: MargotLeadbetter <──┐ │ │
  └─ component: JerryLeadbetter <─┐ │ │ │
                                  │ │ │ │
                           imported components
                                  │ │ │ │
# In LeadbetterFamily.cellml:     │ │ │ │
model: LeadbetterFamilyModel      │ │ │ │
  ├─ component: husband ──────────┘ │ │ │
  └─ component: wife ───────────────┘ │ │
                                      │ │
# In GoodFamily.cellml:               │ │
model: GoodFamilyModel                │ │
  ├─ component: wife ─────────────────┘ │
  └─ component: husband ────────────────┘
```

With the correct font, it will look fine:

<img width="599" alt="Screen Shot 2020-04-08 at 6 23 02 PM" src="https://user-images.githubusercontent.com/602265/78751285-05710580-79c6-11ea-99eb-8c22d4b6d02e.png">

Hmm, I am getting the feeling that you wanted to use some kind of dashed lines to show the imports? I guess I would be happy with either plain or dashed lines. It's just that the dashed lines looked a bit strange to me when I first saw them.

Reply:
Yes, the dashed lines are all imports, the solids are all not.

Reply:
That's fair enough.